src/DeviceInput/deviceInputSystem.ts

```diff
@@ -478,13 +478,16 @@ export class DeviceInputSystem implements IDisposable {
             // Handle Active Touches
             if (this.isDeviceAvailable(DeviceType.Touch)) {
                 const pointer = this._inputs[DeviceType.Touch];
+
                 for (let i = 0; i < pointer.length; i++) {
-                    if (pointer[i][PointerInput.LeftClick] === 1) {
+                    if (pointer[i] && pointer[i][PointerInput.LeftClick] === 1) {

```

Comment:
If you wanted to impress your friends, you could write:
```typescript
    if (pointer[i]?.[PointerInput.LeftClick] === 1) {
```
I have just learnt that the optional chaining syntax also works with array indexing, so I thought I should share my discovery ;)

Reply:
Ooh, I appreciate this discovery.  I wasn't sure if there was an optional chaining syntax for multidimensional arrays.

Reply:
Modified to use optional chaining